grafana-plugin/src/pages/insights/Insights.tsx

```diff
@@ -72,8 +72,12 @@ const Insights = observer(() => {
         setDatasource(`${text}`);
       });
     return () => {
-      stackListener?.unsubscribe();
-      dataSourceListener?.unsubscribe();
+      if (stackListener?.unsubscribe) {
+        stackListener.unsubscribe();
+      }
+      if (dataSourceListener?.unsubscribe) {
+        dataSourceListener.unsubscribe();
```

Comment:
`dataSourceListener?.unsubscribe?.();`

Reply:
Up to you, but you can just reduce each of them to 1 liner each

Reply:
That's a good suggestion, thank you, applied it:)